src/iterative_ensemble_smoother/utils.py

```diff
@@ -9,35 +9,19 @@
 from ._ies import InversionType
 
 
-def _compute_AA_projection(A: npt.NDArray[np.double]) -> npt.NDArray[np.double]:
-    """A^+A projection is necessary when the parameter matrix has fewer rows than
-    columns, and when the forward model is non-linear. Section 2.4.3
-    """
-    _, _, vh = np.linalg.svd(A - A.mean(axis=1, keepdims=True), full_matrices=False)
-    projection: npt.NDArray[np.double] = vh.T @ vh
-    return projection
-
-
 def _validate_inputs(
     response_ensemble: npt.NDArray[np.double],
-    parameter_ensemble: npt.NDArray[np.double],
     noise: Optional[npt.NDArray[np.double]],
     observation_errors: npt.NDArray[np.double],
     observation_values: npt.NDArray[np.double],
 ) -> None:
-
     if len(response_ensemble.shape) != 2:
         raise ValueError(
             "response_ensemble must be a matrix of size (number of responses by number of realizations)"
         )
```

Comment:
Could we optionally validate `param_ensemble`?

```suggestion
    observation_values: npt.NDArray[np.double],
    param_ensemble: Optional[npt.NDArray[np.double]],
) -> None:
    if len(response_ensemble.shape) != 2:
        raise ValueError(
            "response_ensemble must be a matrix of size (number of responses by number of realizations)"
        )
    
    if param_ensemble is not None and len(param_ensemble.shape) != 2:
        raise ValueError(
            "parameter_ensemble must be a matrix of size (number of parameters by number of realizations)"
        )
```

Reply:
Sure, I've pushed a commit that compares the number of columns of `param_ensemble` with `response_ensemble` and also added the check that you propose here.